### DISCUSSION TITLE
Let Renovate be able to resolve secrets

### DISCUSSION THREAD

**Discussion by markussiebert** [renovatebot/renovate#13238] [Suggest an Idea]
### What would you like Renovate to be able to do?

It would be nice if renovate could resolve secrets from several backends out of the box, as they are needed to perform actions with renovate. So you need your credentials for the gitapi, or some repositories.

It would be great if we could resolve the at runtime from backends like AWS SecretsManager, or SOPS, Azure Stuff?

### If you have any ideas on how this should be implemented, please tell us here.

Not really - at the moment I'm interested if you see some functionality like this in renovate, or not.

For my go projects, I heavily use https://github.com/variantdev/vals as it's only purpose is to resolve secrets :-)

### Is this a feature you are interested in implementing yourself?

Yes

**Comment by viceice** [comment]
You can export an async function from `config.js`, there you can resolve any custom secrets.

**Comment by markussiebert** [comment]
ok, maybe you can explain it a little bit more? Sounds really interesting

**Comment by viceice** [comment]
*config.js*
```js
const got = require('got');
const base = {
  /* renovate config */
  platform: 'gitlab';
};
module.exports = async () => {
  const dyn = await got.get('https://httpbin.org/anything').json();
  // do any other async stuff here
  return { ...base, ...dyn }
};

**Comment by rarkins** [comment]
I've converted this to a discussion and support @viceice's suggestion
